### DISCUSSION TITLE
Could not solve cooperative tradeoff for sample X. Provided upper bound smaller than lower bound

### DISCUSSION THREAD

**Discussion by GeraldineKong** [micom-dev/micom#141] [Q&A] [answered]
Hello! Thank you for developing MICOM, the tutorials and Q&A have been really useful. 

I am trying to apply a diet for mice studies which I have found in a publication and I am running into the error: 
`Could not solve cooperative tradeoff for sample123. This can often be fixed by enabling `presolve`, choosing more permissive atol and rtol arguments, or by checking that medium fluxes are > atol`

So I tried with `cooperative_tradeoff()` for one sample which gave me a more informative error message:
`
>>> com = load_pickle("pickles/sample123.pickle")
>>> com.medium = medium['flux']
>>> medium['flux'].head()
reaction
EX_his_L_m    0.083226
EX_ile_L_m    0.183206
EX_leu_L_m    0.332061
EX_lys_L_m    0.250685
EX_met_L_m    0.086577
Name: flux, dtype: float64
>>> sol = com.cooperative_tradeoff(fraction=0.5)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kongg1/.conda/envs/micom/lib/python3.9/site-packages/micom/community.py", line 797, in cooperative_tradeoff
    return cooperative_tradeoff(
  File "/home/kongg1/.conda/envs/micom/lib/python3.9/site-packages/micom/problems.py", line 92, in cooperative_tradeoff
    com.variables.community_objective.ub = min_growth
  File "/home/kongg1/.conda/envs/micom/lib/python3.9/site-packages/optlang/interface.py", line 227, in ub
    raise ValueError(
ValueError: The provided upper bound -1.38936e-14 is smaller than the lower bound -6.94679e-15 of variable -6.946790637316429e-15 <= community_objective <= 1e+20.`

Can you please tell me what I need to do to grow my pickles? Sorry if this is a dumb question, I'm still new to the metabolic modelling world :)

**Comment by cdiener** [comment, marked as answer]
Sorry about that. First, this is a bug that is already fixed in some updates I have lined up. Maybe more important is the cause though. In your case, you are seeing this because no taxon in the microbial community can grow (they all have growth rates indistinguishable from zero). 

When using a medium from somewhere else, this usually happens because you are missing one or more essential cofactors. So one usually has to complete the medium first so the chosen taxa can grow on it. MICOM does come with helpers to do so (and an even easier method will follow soon). 

Documentation for this is still not great but hopefully looking at the recipes in the [media repository](https://github.com/micom-dev/media) can give you a starting point. For instance, you could look at the later steps taken [here](https://github.com/micom-dev/media/blob/main/recipes/himalaya.ipynb).
